server/@types/viewModels/index.d.ts

```diff
@@ -242,8 +242,10 @@ declare module 'viewModels' {
    * DPS frontend components
    */
   export interface FrontendComponentsPageAdditions {
+    headerHtml: string
```

Comment:
I know this story is not related to the header (we have a different story for that), but because of the changes I made to the service and mapper (see below) it made sense to add it

Also added the problemRetrievingData property, consistent with our other types like this